src/screens/TopicPageScreen/elements/BottomSheetFollow.tsx

```diff
@@ -0,0 +1,307 @@
+              marginRight: 8
+            }}>
+            {type === 'signed' ? (
+              <ProfilePicture
+                size={20}
+                width={0}
+                profilePicPath={profilePicture}
+                anonBackgroundColor=""
+                anonEmojiCode=""
+              />
+            ) : (
+              <IconIncognito color={COLORS.white} />
+            )}
+          </View>
+          <Text
+            style={{
+              fontFamily: fonts.inter[400],
+              fontSize: 14,
+              color: COLORS.black
+            }}>
+            {type === 'signed' ? `as @${username}` : 'Incognito'}
+          </Text>
+        </View>
+        <Text
+          style={{
+            fontFamily: fonts.inter[400],
+            fontSize: 12,
+            color: COLORS.blackgrey,
+            marginTop: 4,
+            lineHeight: 18
+          }}>
+          {type === 'signed'
+            ? 'You’ll be visible as a community member.'
+            : 'You’ll be a hidden member of the community.'}
+        </Text>
+      </View>
+      {isFollowing && (
+        <IconCheckCircleOutline
+          color={type === 'signed' ? COLORS.signed_primary : COLORS.anon_primary}
+        />
+      )}
+    </TouchableOpacity>
+  );
+};
+
+const BottomSheetFollow = forwardRef((props: BottomSheetFollowProps, ref: Ref<RBSheet>) => {
+  const {
+    topicName,
+    memberCount,
+    setMemberCount,
+    isFollow,
+    setIsFollow,
+    followType,
+    setFollowType,
+    getTopicDetail,
+    onClose
+  } = props;
+  const {followTopic} = useChatClientHook();
+  const handleFollowTopic = async ({type}: {type: Follow}) => {
+    if (isFollow) {
+      if (followType === '') {
+        setMemberCount(memberCount - 1);
+      }
+    } else {
+      setMemberCount(memberCount + 1);
+    }
+    onClose();
+    if (type === 'signed') {
+      if (followType === '' || followType === 'incognito') {
+        setFollowType('signed');
+        setIsFollow(true);
+        try {
+          await followTopic(topicName, false);
+          getTopicDetail(topicName);
+        } catch (error) {
+          if (__DEV__) {
+            console.log(error);
+          }
+        }
+      }
+    } else if (type === 'incognito') {
+      if (followType === '' || followType === 'signed') {
+        setFollowType('incognito');
+        setIsFollow(true);
+        try {
+          await followTopic(topicName, true);
+          getTopicDetail(topicName);
+        } catch (error) {
+          if (__DEV__) {
+            console.log(error);
+          }
+        }
+      }
+    }
+  };
+
+  const handleUnfollowTopic = async ({type}: {type: Follow}) => {
+    setFollowType('');
+    setIsFollow(false);
+    setMemberCount(memberCount - 1);
+    onClose();
+    try {
+      const isIncognito = type === 'incognito';
+      await followTopic(topicName, isIncognito);
+    } catch (error) {
+      if (__DEV__) {
+        console.log(error);
+      }
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the above comment, the `handleUnfollowTopic` function is also performing multiple tasks. It's setting the `followType` to an empty string, setting `isFollow` to false, decrementing the `memberCount`, and calling the `followTopic` function. It would be better to separate these tasks into distinct functions for better readability and maintainability.

```diff
-  const handleUnfollowTopic = async ({type}: {type: Follow}) => {
-    setFollowType('');
-    setIsFollow(false);
-    setMemberCount(memberCount - 1);
-    onClose();
-    try {
-      const isIncognito = type === 'incognito';
-      await followTopic(topicName, isIncognito);
-    } catch (error) {
-      if (__DEV__) {
-        console.log(error);
-      }
-    }
-  };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->